socks.h

```diff
@@ -76,7 +76,7 @@ inline void sock_close(SOCKET s)
 inline const char* sock_strerror(char* buf, size_t len)
 {
 	buf[0] = '\0';
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || !defined(_GNU_SOURCE) || !defined(__GLIBC__)
```

Comment:
This `|| !defined(_GNU_SOURCE) || !defined(__GLIBC__)` look like it is a little bit to much. The change means that all compiler those are not using `GLIBC` enters this pass. Befor ethis PR only `__APPLE__` enter the pass.
I think  it should be `#if  ( defined(__APPLE__) || defined(__FreeBSD__) ) & ( !defined(_GNU_SOURCE) || !defined(__GLIBC__) )`

My suggestion should be side effect free to other OS than `APPLE` and `FreeBSD`